docs/book/beyond_basics/controller_metrics.md

```diff
@@ -0,0 +1,101 @@
+# HELP controller_runtime_reconcile_queue_length Length of reconcile queue per controller
+# TYPE controller_runtime_reconcile_queue_length gauge
+controller_runtime_reconcile_queue_length{controller="mysql-controller"} 0
+# HELP controller_runtime_reconcile_time_seconds Length of time per reconcile per controller
+# TYPE controller_runtime_reconcile_time_seconds histogram
+controller_runtime_reconcile_time_seconds_bucket{controller="mysql-controller",le="0.005"} 10
+controller_runtime_reconcile_time_seconds_bucket{controller="mysql-controller",le="0.01"} 10
+controller_runtime_reconcile_time_seconds_bucket{controller="mysql-controller",le="0.025"} 10
+controller_runtime_reconcile_time_seconds_bucket{controller="mysql-controller",le="10"} 10
+controller_runtime_reconcile_time_seconds_bucket{controller="mysql-controller",le="+Inf"} 10
+controller_runtime_reconcile_time_seconds_sum{controller="mysql-controller"} 2.3416e-05
+controller_runtime_reconcile_time_seconds_count{controller="mysql-controller"} 10
+# HELP go_gc_duration_seconds A summary of the GC invocation durations.
+# TYPE go_gc_duration_seconds summary
+go_gc_duration_seconds{quantile="0"} 7.69e-05
+go_gc_duration_seconds{quantile="0.25"} 0.0001225
+go_gc_duration_seconds{quantile="0.5"} 0.000124351
+go_gc_duration_seconds{quantile="0.75"} 0.000236344
+go_gc_duration_seconds{quantile="1"} 0.000262102
+go_gc_duration_seconds_sum 0.000822197
+go_gc_duration_seconds_count 5
+# HELP go_goroutines Number of goroutines that currently exist.
+# TYPE go_goroutines gauge
+go_goroutines 39
+# HELP go_info Information about the Go environment.
+# TYPE go_info gauge
+go_info{version="go1.9.4"} 1
+# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
+.....
+....
+
+```
+
+Is the metrics endpoint protected ?
+-----------------------------------
+Yes. By default, kubebuilder generated YAML manifests (under `config/` dir) 
+ensures that the access to metrics endpoint is authenticated and authorized using
+a [auth proxy](https://github.com/brancz/kube-rbac-proxy) which is deployed as 
+sidecar container in the manager pod. You can read more details about the 
+auth proxy based approach [here](https://brancz.com/2018/02/27/using-kube-rbac-proxy-to-secure-kubernetes-workloads/).
+
+If you want to disable the auth proxy, which is not reccommended, you can follow
+the instructions below:
+```sh
+
+# instructions to disable the auth proxy
+TODO (mengqi):
+
+```
+
+How do I configure Prometheus Server to access the metrics?
+-----------------------------------------------------------
+
+Kubebuilder generated manifests for manager have annotations such as
+`prometheus.io/scrape`, `prometheus.io/path` on the metrics service so
+that it can be easily discovered by the prometheus server deployed in your
+kubernetes cluster. 
+
+Assuming auth is enabled, which is by default, you will have to add the
+following to the job which is configured to scrap kubernetes endpoints.
```

Comment:
I'd use `kubernetes service endpoints` instead of `kubernetes endpoints`.
IMHO using `kubernetes service endpoints` is less confusing, since it is what prometheus uses in its comments :)